### PR TITLE
added a loader when loading readme files

### PR DESCRIPTION
**Pull request by RizkyRajitha**
added a loader

### CODE REVIEW

File: src/assets/styles/loader.css [hunk 1]
```diff
@@ -0,0 +1,184 @@
+.loaderinnerdiv {
+  margin: 0 auto;
+  margin-top: 20%;
+  /* padding: 20%; */
+  /* width: 200%;
+  height: 200%; */
```
Comment (carolinaknoll) [inline src/assets/styles/loader.css]: Please remove comments. 😄

File: src/components/topic-list.js [hunk 2]
```diff
@@ -1,58 +1,75 @@
...
-    axios.get(readmeUrl)
-    .then((topicMarkdown) => {
-      this.setState({
-        topicMarkdown: topicMarkdown.data,
-        errorMessage: '',
-      });
-
-      if (!this.state.topicMarkdown) {
+  getRawAwesomeListContent = readmeUrl => {
+    // console.log("loading...");
```
Comment (carolinaknoll) [inline src/components/topic-list.js]: Please remove console.log. 😄

File: src/components/topic-list.js [hunk 3]
```diff
@@ -1,58 +1,75 @@
...
+    this.setState({ isloading: true });
+    axios
+      .get(readmeUrl)
+      .then(topicMarkdown => {
+        this.setState({
+          topicMarkdown: topicMarkdown.data,
+          errorMessage: "",
+          isloading: false
+        });
+        // console.log("listconxios");
```
Comment (carolinaknoll) [inline src/components/topic-list.js]: Please remove console.log. 😄

File: src/components/topic-list.js [hunk 4]
```diff
@@ -1,58 +1,75 @@
...
-    .then((res) => {
-      const { data: { download_url } } = res;
-      return this.getRawAwesomeListContent(download_url);
-    })
-    .catch((error) => {
-      this.setState({
-        errorMessage: `There was an error. Unable to load the Awesome list for ${this.props.clickedTopic}. ${error}.`,
-        topicMarkdown: ''
+    this.setState({ isloading: true });
+    // console.log("redme");
```
Comment (carolinaknoll) [inline src/components/topic-list.js]: Please remove console.log. 😄

File: src/components/topic-list.js [hunk 5]
```diff
@@ -1,58 +1,75 @@
...
+    axios
+      .get(`https://api.github.com/repos/${this.props.clickedTopic}/readme`)
+      .then(res => {
+        // console.log("done...");
```
Comment (carolinaknoll) [inline src/components/topic-list.js]: Please remove console.log. 😄

File: src/components/topic-list.js [hunk 6]
```diff
@@ -163,24 +197,34 @@ export default class TopicList extends Component {
       let parsedMarkdown = customMarked(this.state.topicMarkdown);
       parsedMarkdown = this.addButtonsToListElements(parsedMarkdown);
 
+      // if (this.state.isloading) {
+      //   return <Loader />
+      // } else if (this.state.isloading === false) {
```
Comment (carolinaknoll) [inline src/components/topic-list.js]: Please remove this commented code. 😄

File: src/components/topic-list.js [hunk 7]
```diff
@@ -163,24 +197,34 @@ export default class TopicList extends Component {
...
+  };
 
-	render() {
+  render() {
     return (
       <div className="topic-list-container">
-      {this.props.clickedTopic ? (
-        this.renderTopicList()
-      ) : null}
+        {/* {this.state.isloading ? <Loader /> : ""} */}
```
Comment (carolinaknoll) [inline src/components/topic-list.js]: Please remove this commented code. 😄

**Review by carolinaknoll — CHANGES_REQUESTED** [review]
Hello @RizkyRajitha this looks very nice, you did a great job! 🚀 

However, I've pointed out a few comments on your code here on your PR so that you can remove commented code and console.logs.

Please remove these and commit it so that I can merge your PR. 😄 

Thank you!

**Comment by RizkyRajitha** [comment]
ok @carolinaknoll  i will remove commented code blocks

**Comment by RizkyRajitha** [comment]
i made the necessary changes and also figured out how to add space on loader . and committed to this branch .
thank you @carolinaknoll

**Review by carolinaknoll — APPROVED** [review]

**Comment by carolinaknoll** [comment]
This is awesome, thank you so much for contributing! 🚀

**Pull request merged**